master/buildbot/steps/source/base.py

```diff
@@ -220,6 +220,36 @@ def evaluateCommand(_):
             return cmd.rc
         return d
 
+    def _downloadFile(self, buf, filename, workdir=None, mode=None):
```

Comment:
There is the CompositeStepMixin, which provide this method

Reply:
Yeah, noticed that. I just wanted to leave refactoring of this for another PR.